[DOCS][ResponseOps][9.5 & Serverless] Document Elastic Cloud identifier in connector User-Agent headers (#268745)
elastic/kibana: docs/reference/connectors-kibana.md

:::

> ## Elastic Cloud identifier in connector `User-Agent` headers [connector-user-agent-cloud]

Reviewer: Can we lead with the goal of what this is helping with? To make it more parallel with the rest of the page

Author: Ah yeah, good catch. Fixed in [`b75ac9f` (this PR)](https://github.com/elastic/kibana/pull/268745/commits/b75ac9fd2c1c976e84dd79e7a2f3b1b999dd6ac4).

```{applies_to}
stack: ga 9.5+
serverless: ga
deployment:
  ess: ga
  self: unavailable
```

:::{important}

> When {{kib}} runs on {{ech}} or Elastic Cloud Serverless, outbound connector HTTP requests include an Elastic Cloud deployment or serverless project identifier in the `User-Agent` header. On self-managed or on-premises deployments, connector requests **do not** add this fragment. The `User-Agent` stays at the default HTTP client value (for example, `axios/1.7.2`).

Reviewer: ```suggestion
When {{kib}} runs on {{ech}} or Elastic Cloud Serverless, outbound connector HTTP requests include a deployment or project identifier in the `User-Agent` header. On self-managed or on-premises deployments, connector requests **do not** add this fragment. The `User-Agent` stays at the default HTTP client value (for example, `axios/1.7.2`).
```
Same as other comment.

+what do we mean by self-managed here? ECE and ECK? In our docs, self-managed means on-prem, non-orchestrated currently. So we better be explicit if we're referring to these deployment types.

Author: @mikecote @seanstory would you be able to help here? Do self-managed deployments include Elastic Cloud Enterprise and Elastic Cloud on Kubernetes?

Reviewer 2: Hmm, I am not sure, ECH for sure but for ECE/ECK it depends if they provide `xpack.cloud*` configuration settings. Asking Cursor, it seems ECE might but not ECK.

Reviewer 3: Good question. I'm not super familiar with either, but from some brief Clauding, it looks like ECE _will_ contain a deployment_id. ECK will not.

Evidence:
- ECE role [used here](https://github.com/elastic/cloud/blob/b015a8cec4755e8833aa7a29767c47a6a20efa82/scala-services/adminconsole/src/test/scala/no/found/adminconsole/api/v1/deployments/services/controlplanesettings/providers/kibana/KibanaCloudUrlSettingsSpec.scala#L53)
- expects `deployment_url` to be set [here](https://github.com/elastic/cloud/blob/b015a8cec4755e8833aa7a29767c47a6a20efa82/scala-services/adminconsole/src/test/scala/no/found/adminconsole/api/v1/deployments/services/controlplanesettings/providers/kibana/KibanaCloudUrlSettingsSpec.scala#L112)

Author: Thanks, all. I modified the wording in the note to be the callout to be consistent with the way deployment types are referenced in the new secton that this PR adds. The note now explains that when the header is set based on whether cloud deployment or project metadata is available (rather than “self-managed”), which covers Elastic Cloud Hosted, Serverless, and Elastic Cloud Enterprise when the orchestrator injects cloud metadata. Self-managed and Elastic Cloud on Kubernetes deployments keep the default `User-Agent`.

:::

> For every outbound HTTP request made by a connector (for example, to Slack, Google Workspace, or PagerDuty), {{kib}} sets a `User-Agent` header that includes an identifier for the Elastic Cloud deployment or serverless project that originated the traffic. This is automatic and does not require connector configuration, networking settings, or other administrator actions.

Reviewer: ```suggestion
For every outbound HTTP request made by a connector (for example, to Slack, Google Workspace, or PagerDuty), {{kib}} sets a `User-Agent` header that includes an identifier for the deployment or project that originated the traffic. This is automatic and does not require connector configuration, networking settings, or other administrator actions.
```
better use just deployment or project otherwise you have to stay "parallel" every time you mention either hosted deployments or serverless projects, or when using Elastic Cloud

Author: This reminds me that I need to add style guidelines to my agent.md file so it doesn't keep doing this.

IT and information security teams can use this value to correlate traffic recorded in third-party audit or access logs with a specific Elastic Cloud deployment or project. For example, when investigating incidents or working with a vendor that contacts Elastic about suspicious activity.

### Header format

The header starts with the underlying HTTP client name and version (for example, `axios/1.7.2`). On {{ech}} and Elastic Cloud Serverless, an additional Elastic fragment is appended.

| Environment | Example suffix in `User-Agent` | Meaning |
| --- | --- | --- |
| {{ech}} | `elastic (deployment:<deployment_id>)` | `<deployment_id>` is the deployment ID. |
| Elastic Cloud Serverless | `elastic (project:<project_id>)` | `<project_id>` is the serverless project ID. |

> {{kib}} uses `project:<project_id>` when Cloud metadata includes a serverless project ID; otherwise it uses `deployment:<deployment_id>` when a deployment ID is present. Hosted and serverless environments normally expose only one of these identifiers.

Reviewer: Feels confusing as it gives the impression that it's adding information while it basically repeats the table in a less clear way
```suggestion
```

Author: That's fair - removing this doesn't detract from the docs.

### Look up the deployment or project from an identifier

> Looking up these identifiers is limited to Elastic Cloud organization administrators. There is no {{kib}} UI or API to resolve the values shown in connector `User-Agent` headers.

Reviewer: > Looking up these identifiers is limited to Elastic Cloud organization administrators.

Are we saying here that only org admins can go to the pages listing deployments and projects? Phrased this way it could be interpreted as that or as only org admins can retrieve these IDs. 

Once we figure it out we can also put it in active voice I think.

Author: Fixed in [`b75ac9f` (this PR)](https://github.com/elastic/kibana/pull/268745/commits/b75ac9fd2c1c976e84dd79e7a2f3b1b999dd6ac4).

1. Copy the identifier from the header. The identifier is the substring after `deployment:` or `project:`.
2. Open the [Elastic Cloud console](https://cloud.elastic.co) and sign in.

> 3. Go to the management pages for deployments or projects, and enter the ID in the search field to find the matching deployment or project.

Reviewer: ```suggestion
3. Go to the **Hosted** (for {{ech}} deployments) or **Serverless** (for Serverless projects) page, and enter the ID in the search field to find the matching deployment or project.
```
"management pages" isn't very clear

Author: fine!
